projects/netgrif-components-core/src/lib/authentication/services/methods/basic-authentication/basic-authentication.service.ts

```diff
@@ -28,7 +28,8 @@ export class BasicAuthenticationService extends AuthenticationMethodService {
         }
 
         return this._http.get<UserResource>(url, {
-            headers: new HttpHeaders().set('Authorization', `Basic ${btoa(`${credentials.username}:${credentials.password}`)}`)
+            headers: new HttpHeaders().set('Authorization', 'Basic ' +
+        `${btoa(unescape(encodeURIComponent(`${credentials.username}:${credentials.password}`)))}`)
```

Comment:
the `unescape` method should no longer be used according to the docs https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/unescape

according to the docs, the `unescape` method is similar to the `decodeURIComponent` method, which should be the inverse operation to `encodeURIComponent` used here. What benefit does encoding and then decoding the string yield?

Does this change not affect the decoding of the token at the backend?

This change should be covered by some new unit tests in my opinion.

Reply:
It is strange behaviour of btoa in google chrome. First needs to be encoded to hexa then decoded back to normal and after that it accepts the encoded values. I'll replace unescape with encodeUriComponent. Decoding on backend was checked, it is not affected. I will implement a unit test.